core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/SignForIdentifierHandler.java

```diff
@@ -59,11 +67,21 @@ public void handle(final RoutingContext routingContext) {
       signerForIdentifier
           .sign(normaliseIdentifier(identifier), data)
           .ifPresentOrElse(
-              signature ->
+              signature -> {
+                if (slashingProtection.isPresent()) {
+                  if (slashingProtection.get().maySignAttestation(null, null, null)) {
```

Comment:
how do we check that block signing is allowed?

Reply:
we also have an API for that, but can;t use it - as we don't know if its a block or attestation yet

Reply:
yeah good point